client/blocks/reader-post-card/index.jsx

```diff
@@ -115,99 +112,74 @@ export default class ReaderPostCard extends React.Component {
+			showMenuFollow={ isDiscover }
+			onCommentClick={ onCommentClick }
+			showEdit={ false }
+			className="ignore-click"
+			iconSize={ 18 } />;
+
+		let readerPostCard;
+		if ( isPhotoPost ) {
+			const { height, width } = post.canonical_media;
+			readerPostCard = <PhotoPost imageUri={ post.canonical_media.src }
+				href={ post.URL }
+				imageSize={ { height, width } }
+				onExpanded={ this.handlePhotoCardExpanded }
+				title={ title } >
+					{ discoverFollowButton }
+					{ readerPostActions }
+				</PhotoPost>;
+		} else if ( isGalleryPost ) {
+			readerPostCard = <GalleryPost post={ post } title={ title } >
+					{ readerPostActions }
+				</GalleryPost>;
 		} else {
-			featuredAsset = isPhotoOnly
-				? <PostPhoto imageUri={ post.canonical_media.src } href={ post.URL } imageSize={ {
-					height: post.canonical_media.height,
-					width: post.canonical_media.width,
-				} } onExpanded={ this.handlePhotoCardExpanded } />
-				: <FeaturedImage imageUri={ post.canonical_media.src } href={ post.URL } />;
+			readerPostCard = <StandardPost post={ post } title={ title } >
+					{ isDailyPostChallengeOrPrompt( post ) && <DailyPostButton post={ post } tagName="span" /> }
+					{ discoverFollowButton }
+					{ readerPostActions }
+				</StandardPost>;
 		}
 
+		const followUrl = feed ? feed.feed_URL : post.site_URL;
+
 		return (
 			<Card className={ classes } onClick={ this.handleCardClick }>
 				<PostByline post={ post } site={ site } feed={ feed } showSiteName={ showSiteName } />
 				{ showPrimaryFollowButton && followUrl && <FollowButton siteUrl={ followUrl } followSource={ followSource } /> }
-				<div className="reader-post-card__post">
-					{ ! isGallery && featuredAsset }
-					{ isGallery && <PostGallery post={ post } /> }
-					<div className="reader-post-card__post-details">
-						<AutoDirection>
-							<h1 className="reader-post-card__title">
-								<a className="reader-post-card__title-link" href={ post.URL }>{ title }</a>
-							</h1>
-						</AutoDirection>
-						{ showExcerpt && (
-								<AutoDirection>
-									<div className="reader-post-card__excerpt"
-										dangerouslySetInnerHTML={ { __html: post[ excerptAttribute ] } } // eslint-disable-line react/no-danger
-									/>
-								</AutoDirection> )
-						}
-						{ isDailyPostChallengeOrPrompt( post ) && <DailyPostButton post={ post } tagName="span" /> }
-						{ discoverBlogName &&
-							<DiscoverFollowButton
-									siteName={ discoverBlogName }
-									followUrl={ DiscoverHelper.getSourceFollowUrl( post ) } />
-						}
-
-						{ post &&
-							<ReaderPostActions
-								post={ originalPost ? originalPost : post }
-								visitUrl = { post.URL }
-								showVisit={ true }
-								showMenu={ true }
-								showMenuFollow={ ! isDiscoverPost }
-								onCommentClick={ onCommentClick }
-								showEdit={ false }
-								className="ignore-click"
-								iconSize={ 18 } />
-						}
-					</div>
-				</div>
+				{ readerPostCard }
```

Comment:
Good point, I think you're right on the child prop. I was trying to stay as true to the original component JSX as possible which did include the `children` prop. But at this level of abstraction it doesn't serve the same purpose.